src/linear_algebra/linear_independent.lean

```diff
@@ -446,6 +446,38 @@ theorem linear_independent.image {ι} {s : set ι} {f : ι → M}
   (hs : linear_independent R (λ x : s, f x)) : linear_independent R (λ x : f '' s, (x : M)) :=
 by convert linear_independent.image_of_comp s f id hs
 
+lemma linear_independent.group_smul
+  {G : Type*} [hG : group G] [distrib_mul_action G R] [distrib_mul_action G M]
+  [is_scalar_tower G R M] [smul_comm_class G R M] {v : ι → M} (hv : linear_independent R v)
+  (w : ι → G) : linear_independent R (w • v) :=
+begin
+  rw linear_independent_iff'' at hv ⊢,
+  intros s g hgs hsum i,
+  refine (smul_eq_zero_iff_eq (w i)).1 _,
+  refine hv s (λ i, w i • g i) (λ i hi, _) _ i,
+  { dsimp only,
+    exact (hgs i hi).symm ▸ smul_zero _ },
+  { rw [← hsum, finset.sum_congr rfl _],
+    intros, erw [pi.smul_apply, smul_assoc, smul_comm] },
+end
+
+-- This lemma cannot be proved with `smul_group_linear_independent` since the action of
+-- `units R` on `R` is not commutative.
```

Comment:
I think the generalization we're missing here is maybe about right actions. I think the key step in the proof is solving
```lean
(g x •← w x) • v x = g x • w x • v x
```
where ``notation a ` •← ` b := op b • a``, and the action `r •← u` is defined using #7723 as
```lean
instance (R) [semiring R] : distrib_mul_action (units R)ᵒᵖ R :=
distrib_mul_action.comp_hom R (units.op_equiv.symm : (units R)ᵒᵖ ≃* _).to_monoid_hom
```
To complete this generalization, we'd need a variable of `smul_assoc` that covers `(g x •← w x) • v x = g x • w x • v x`, which is definitely out of scope for this PR.